commands/project/dev.ts

```diff
@@ -107,16 +110,20 @@ exports.handler = async options => {
     (!hasPublicApps && isSandbox(accountConfig));
 
   // The account that the project must exist in
-  let targetProjectAccountId = derivedAccountId;
+  let targetProjectAccountId = providedAccountId
+    ? parsedProvidedAccountId
+    : null;
   // The account that we are locally testing against
-  let targetTestingAccountId = derivedAccountId;
+  let targetTestingAccountId = providedAccountId
+    ? parsedProvidedAccountId
+    : null;
 
   // Check that the default account or flag option is valid for the type of app in this project
-  if (derivedAccountId) {
-    checkIfAccountFlagIsSupported(accountConfig, hasPublicApps);
+  if (providedAccountId) {
```

Comment:
Won't `providedAccountId` always be the same as `derivedAccountId` when the user passes in `--account`? Derived will first look for the account flag, and then fall back to using the default account if one isn't specified.

Reply:
No, it won't. `derivedAccountId` will always be defined in commands, since we inject the default value into it in middleware. `providedAccountId` is only defined when a user passes in an account flag, and that's important here. We want different behavior depending on if the user passes in a flag.

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1303/commits/1b2ade7d8896ec96bd1e86447be2d8c4a7ae269e

Reply:
Ohhh right, because we bypass the prompts when the user passes in the flag. I forgot about that 👌